src/components/Global/HistogramSlider.vue

```diff
@@ -0,0 +1,229 @@
+      }
+      else {
+        previousDistance = distance
+      }
+    }
+    return props.bins.length - 1
+  }
+}
+
+function updateScaleRange() {
+  // This is called when the range slider control is moved
+  scaleRange.value[0] = sliderToBinValue(sliderRange.value[0])
+  scaleRange.value[1] = sliderToBinValue(sliderRange.value[1])
+  emit('updateScaling', scaleRange.value[0], scaleRange.value[1])
+}
+
+function updateLowerScale(value) {
+  // This is called when a change is made on the lower point number control
+  scaleRange.value = [Number(value), scaleRange.value[1]]
+  sliderRange.value[0] = scaleToSliderValue(Number(value))
+  emit('updateScaling', scaleRange.value[0], scaleRange.value[1])
+}
+
+function updateUpperScale(value) {
+  // This is called when a change is made on the upper point number control
+  scaleRange.value = [scaleRange.value[0], Number(value)]
+  sliderRange.value[1] = scaleToSliderValue(Number(value))
+  emit('updateScaling', scaleRange.value[0], scaleRange.value[1])
+}
+
+function zScaleImage() {
+  // This is called to reset the ranges to the zMin/zMax
+  scaleRange.value = [props.zMin, props.zMax]
+  sliderRange.value = [scaleToSliderValue(props.zMin), scaleToSliderValue(props.zMax)]
+  emit('updateScaling', scaleRange.value[0], scaleRange.value[1])
+}
+
+watch(
+  () => props.zMax, () => {
+    zScaleImage()
+  }
+)
+
+</script>
+<template>
+  <v-row class="ml-4">
+    <v-col cols="4">
+      <v-text-field
+        v-model="scaleRange[0]"
+        :min="0"
+        label="lower point"
+        density="compact"
+        type="number"
+        variant="outlined"
+        step="10"
+        hide-details
+        @update:modelValue="updateLowerScale">
+      </v-text-field>
+    </v-col>
+    <v-col cols="4">
+      <v-text-field
+        v-model="scaleRange[1]"
+        :max="props.maxValue"
+        label="upper point"
+        density="compact"
+        type="number"
+        variant="outlined"
+        step="10"
+        hide-details
+        @update:modelValue="updateUpperScale">
+      </v-text-field>
+    </v-col>
+    <v-col cols="4">
+      <v-btn
+        class="ml-2"
+        variant="outlined"
+        @click="zScaleImage"
+      >Z-Scale</v-btn>
+    </v-col>
+  </v-row>
+  <v-row class="ma-0">
+    <v-sheet class="stackSheet">
+      <v-sparkline
+        smooth=true
+        fill=true
+        line-width="0.1"
+        height="50"
+        :gradient="gradient"
+        gradient-direction='left'
+        :model-value="props.histogram"
+        auto-draw
+      >
+      </v-sparkline>
+      <v-range-slider
+        class="stackControls"
+        v-model="sliderRange"
+        step="1"
+        track-size="0"
+        :track-color="backgroundColor"
+        :track-fill-color="selectedColor"
+        :thumb-color="selectedColor"
+        thumb-size="16"
+        :max="props.bins.length-1"
+        thumb-label="always"
+        strict
+        hide-details
+        @update:modelValue="updateScaleRange"
```

Comment:
Don't need to change it, but wanted to check this to help me learn more about vue. A watch function on slider Range would be equivalent to listening for an update event of the components model Value. Only benefit is someone reading it that doesn't know vue as well may be able to understand watchSliderRange vs making the connection that the variable in v-model is what is triggering the @ update:modelValue event.

Reply:
So there can actually be a difference in behavior between those two things in practice. Setting an on update handler in the component will only get triggered when the value changes due to the component, i.e. by moving the slider. This is different from a watcher, that will get triggered any time the value changes by any means, including just setting it elsewhere in the code. So you'll see I do slightly different logic based on if the value is changed from the range slider vs from the number inputs, even though both of those things will ultimately change the value in the variable.